src/mach/load_command.rs

```diff
@@ -507,25 +507,27 @@ impl<'a> ctx::TryFromCtx<'a, Endian> for ThreadCommand {
         let flavor: u32 = bytes.pread_with(8, le)?;
         let count: u32 = bytes.pread_with(12, le)?;
 
+        if count > 70 {
+            return Err(error::Error::Malformed(format!(
+                "thread command specifies {} longs for thread state but we handle only 70",
+                count
+            )));
+        }
+
         // get a byte slice of the thread state
         let thread_state_byte_length = count as usize * 4;
-        let thread_state_bytes = &bytes[16..16 + thread_state_byte_length];
 
         // check the length
-        if thread_state_bytes.len() < thread_state_byte_length {
+        if bytes.len() < 16 + thread_state_byte_length {
```

Comment:
i know you didn't introduce 16 but it might be nice to turn this into a `const` since it's used again below and will be puzzling to future reviewers